### PR TITLE
feat(library): Introduce Post Processor for Models

### PR DESCRIPTION
**Pull request by rajeshgangireddy**
# Pull Request

Currently, models produce raw segmentation masks that often contain overlapping predictions, small noisy fragments, and artifacts. Post-processing (NMS, filtering) was previously either hardcoded inside the SAM decoder or absent entirely, making it impossible to customize, chain, or export these operations.

Therefore, this PR introduces a model-agnostic mask post-processing pipeline built as nn.Module components that plug into any model via the Model base class. 
Main Components : 
- PostProcessor : abstract base class; every processor implements forward(masks, scores, labels) → (masks, scores, labels).
- PostProcessorPipeline :  chains processors sequentially; auto-strips non-exportable ones during ONNX export.
- Integration :  Model.__init__ accepts an optional postprocessor param; predict() calls apply_postprocessing() after inference.

Category | Processors | ONNX-safe
-- | -- | --
NMS | MaskNMS, BoxNMS, MaskIoMNMS, BoxIoMNMS, SoftNMS | Yes
Filtering | ScoreFilter, MinimumAreaFilter | Yes
Morphology | MorphologicalOpening, MorphologicalClosing | Yes
Merge | InstanceMerge, MergePerClassMasks | Partial
Overlap | PanopticArgmaxAssignment | Yes
Connected Components | ConnectedComponentFilter, HoleFilling | No (OpenCV/SciPy)

~~Still in draft due to~~
- [x] Try to make every Postprocessing method onnx/openvino exportable
- [x] Cleanup - I have tried too many methods, only a few of them work, remove the rest 
- [x] Some testing scripts are inlcuded - need to cleanup or remove them 
- [x] Add default PostProcessor for all models? 

image

image

## Type of Change

- [x] ✨ `feat` - New feature
- [ ] 🐞 `fix` - Bug fix
- [ ] 📚 `docs` - Documentation
- [ ] ♻️ `refactor` - Code refactoring
- [ ] 🧪 `test` - Tests
- [ ] 🔧 `chore` - Maintenance

## Breaking Changes

---

### CODE REVIEW

File: library/src/instantlearn/components/postprocessing/base.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        >>> pipeline = PostProcessorPipeline([
+        ...     MinimumAreaFilter(min_area=64),
+        ...     MorphologicalOpening(kernel_size=3),
+        ... ])
+    """
+
+    def __init__(self, processors: list[PostProcessor]) -> None:
+        """Initialize the pipeline with an ordered list of processors."""
+        super().__init__()
+        self.processors = nn.ModuleList(processors)
```
Comment (samet-akcay) [inline library/src/instantlearn/components/postprocessing/base.py]: `Postprocessor` is an `nn.Module`. 

`PostprocessorPipeline` contains `list[PostProcessor]`, which is `nn.ModuleList`. Would it be an idea to name `PostprocessorPipeline` as `PostProcessorList` for consistency?
